features/apimgt/org.wso2.carbon.apimgt.store.feature/src/main/resources/devportal/source/src/app/components/Shared/AppsAndKeys/ApiKeyManager.jsx

```diff
@@ -139,60 +178,141 @@ class ApiKeyManager extends React.Component {
+                            defaultMessage="Use the Generate Key button to generate a self-contained JWT token."
+                          />
+                        </Typography>
+                      </Grid>
                     </Grid>
-                </FormGroup>
-                <Dialog open={open} onClose={this.handleClose} aria-labelledby='form-dialog-title'>
-                    <DialogTitle id='responsive-dialog-title' className={classes.dialogTitle}>
-                        {'Generate API Key'}
+                  </FormGroup>
+                  <Dialog
+                    open={open}
+                    onClose={this.handleClose}
+                    aria-labelledby="form-dialog-title"
+                  >
+                    <DialogTitle id="responsive-dialog-title" className={classes.dialogTitle}>
+                      {"Generate API Key"}
                     </DialogTitle>
                     <DialogContent className={classes.dialogContent}>
-                        <DialogContentText>
-                            {!showToken && (
-                                <ApiKey
-                                    updateAccessTokenRequest={this.updateAccessTokenRequest}
-                                    accessTokenRequest={accessTokenRequest}
-                                />
-                            )}
-                            {showToken && <ViewToken token={apikey} />}
-                        </DialogContentText>
-                    </DialogContent>
-                    <DialogActions>
+                      <DialogContentText>
                         {!showToken && (
-                            <Button onClick={() => this.generateKeys(selectedApp, keyType)} disabled={!accessTokenRequest.timeout} color='primary'>
-                                <FormattedMessage
-                                    id='Shared.AppsAndKeys.ViewKeys.consumer.generate.btn'
-                                    defaultMessage='Generate'
-                                />
-                            </Button>
+                          <ApiKey
+                            updateAccessTokenRequest={this.updateAccessTokenRequest}
+                            accessTokenRequest={accessTokenRequest}
+                          />
                         )}
-                        <Button onClick={this.handleClose} color='primary' autoFocus>
-                            <FormattedMessage
-                                id='Shared.AppsAndKeys.ViewKeys.consumer.close.btn'
-                                defaultMessage='Close'
-                            />
+                        {showToken && <ViewToken token={apikey} />}
+                      </DialogContentText>
+                    </DialogContent>
+                    <DialogActions>
+                      {!showToken && (
+                        <Button
+                          onClick={() => this.generateKeys(selectedApp, keyType)}
```

Comment:
This is an anti-pattern, We have created a lambda function and called a class instance method without using any event data ?

Reply:
fixed